packages/js/src/ui/components/Inbox.tsx

```diff
@@ -21,28 +21,40 @@ export type InboxProps = {
   onSecondaryActionClick?: NotificationActionClickHandler;
 };
 
-enum Screen {
-  Inbox = 'inbox',
+export enum InboxPage {
+  Notifications = 'notifications',
   Preferences = 'preferences',
 }
 
-type InboxContentProps = {
+export type InboxContentProps = {
   mountNotification?: NotificationMounter;
   onNotificationClick?: NotificationClickHandler;
   onPrimaryActionClick?: NotificationActionClickHandler;
   onSecondaryActionClick?: NotificationActionClickHandler;
+  initialPage?: InboxPage;
+  hideNav?: boolean;
 };
 
-const InboxContent = (props: InboxContentProps) => {
-  const [currentScreen, setCurrentScreen] = createSignal<Screen>(Screen.Inbox);
+export const InboxContent = (props: InboxContentProps) => {
+  const [currentPage, setCurrentPage] = createSignal<InboxPage>(props.initialPage || InboxPage.Notifications);
   const { tabs, filter } = useInboxContext();
   const style = useStyle();
 
+  const navigateToPage = createMemo(() => (page: InboxPage) => {
```

Comment:
The extra arrow function can be removed. Is there a reason why it needs to return a function ?

```
const navigateToPage = createMemo(page => {
  if(props.hideNav) {
    return
  }
  setCurrentPage(page)
})
```

Reply:
Yes, it's the equivalent of `useCallback` in react.
Essntially `useCallback` is a `useMemo` with an extra arrow function.